[RnD-EasyPurchase-iOS][478620] Implement Payment Queue Controller.

sadat-399 opened the pull request:
- Implement Payment Queue Controller.
- This Controller mainly handled subscription lifecycle events like purchase.
- Ref: https://redmine.bjitgroup.com/redmine/issues/478620

sadat-399 committed:
[RnD-EasyPurchase-iOS][478620] Implement Payment Queue Controller.

- Implement Payment Queue Controller.
- This Controller mainly handled subscription lifecycle events like purchase.

Ref: https://redmine.bjitgroup.com/redmine/issues/478620

RaiyanSharif90 commented on Sources/EasyPurchase/PaymentQueueController.swift:
```diff
@@ -0,0 +1,71 @@
+
+// PaymentQueueController class responsible for managing the payment queue
+class PaymentQueueController: NSObject, PaymentQueue, SKPaymentTransactionObserver {
+    var paymentsController: PaymentsController
```
if the variable is not required to access from external call then make it private

FaiazRahman786 replied:
We do not need `PaymentQueueController` to conform to `PaymentQueue`
If you make a custom `PaymentQueue`, instead of conforming to it, Make an object of it inside the class and use it.

RaiyanSharif90 commented on Sources/EasyPurchase/PaymentQueueController.swift:
```diff
@@ -0,0 +1,71 @@
+}
+
+// PaymentQueueController class responsible for managing the payment queue
+class PaymentQueueController: NSObject, PaymentQueue, SKPaymentTransactionObserver {
```
if this class needs to restrict future inheritance then it should be set as final class

RaiyanSharif90 commented on Sources/EasyPurchase/PaymentQueueController.swift:
```diff
@@ -0,0 +1,71 @@
+//  PaymentQueueController.swift
+//  EasyPurchase
+//
+//  Created by Sadat Ahmed on 20.09.2023.
```
Delete Author name

RaiyanSharif90 commented on Sources/EasyPurchase/PaymentQueueController.swift:
```diff
@@ -0,0 +1,71 @@
+                queue.finishTransaction(transaction)
+
+            default:
+                break
```
comment can be add under default case like 
// Need to handle default case it future

RaiyanSharif90 commented on Sources/EasyPurchase/PaymentQueueController.swift:
```diff
@@ -0,0 +1,71 @@
+    init(paymentsController: PaymentsController = PaymentsController()) {
+        // super.init()
+        self.paymentsController = paymentsController
+        // SKPaymentQueue.default().add(self)
```
you can delete if the code will never require for future development on reference

RaiyanSharif90 commented on Sources/EasyPurchase/PaymentQueueController.swift:
```diff
@@ -0,0 +1,71 @@
+        // SKPaymentQueue.default().add(self)
+    }
+
+    // Start a payment process with the given payment information
```
You can add  // MARK: - startPayment .... for easy segmentation

sadat-399 committed:
[IAP][Func][Review-Feedback]: Implement Payment Queue Controller

Commit ref:
  - Fix the class and variable scope related issue.
  - Change the implementation as per review.
  - Refactor the codebase as well.

Ref: https://redmine.bjitgroup.com/redmine/issues/478620

sadat-399 committed:
[IAP][Func][Review-Feedback]: Implement Payment Queue Controller

Commit ref:
  - Changed the scop of the variable.

Ref: https://redmine.bjitgroup.com/redmine/issues/478620

TulonBJIT committed:
Merge branch 'develop' of ssh://github.com/TulonBJIT/EasyPurchase into Task_478620
